cosmic-app-list/src/app.rs

```diff
@@ -181,8 +181,7 @@ impl DockItem {
 
         let app_icon = AppletIconData::new(applet);
 
-        let cosmic_icon = IconSource::from_unknown(desktop_info.icon().unwrap_or_default())
-            .as_cosmic_icon()
+        let cosmic_icon = cosmic::widget::icon::from_name(desktop_info.icon().unwrap_or_default())
```

Comment:
It is not applying the icon theme, here `Qogir-ubuntu`.
![image](https://github.com/user-attachments/assets/73a5ce15-fbc7-4337-abfe-862eefcf4579)
It could be checked if the path is absolute or IconSource could be improved.

Reply:
Regarding the App Tray showing the wrong icon, it displays the Krita icon when Input Method (im-config) is launched.